sqlmesh/core/environment.py

```diff
@@ -3,55 +3,28 @@
 import json
 import typing as t
 
-from pydantic import validator
+from pydantic import Field, validator
 
+from sqlmesh.core import constants as c
+from sqlmesh.core.config import EnvironmentSuffixTarget
 from sqlmesh.core.snapshot import SnapshotId, SnapshotTableInfo
 from sqlmesh.utils import word_characters_only
 from sqlmesh.utils.date import TimeLike
 from sqlmesh.utils.pydantic import PydanticModel
 
 
-class Environment(PydanticModel):
-    """Represents an isolated environment.
-
-    Environments are isolated workspaces that hold pointers to physical tables.
+class EnvironmentNamingInfo(PydanticModel):
```

Comment:
can we just make this a config option and not pass this around everywhere?

Reply:
ok, i don't really like the name, it's a bit verbose and now it's passed everywhere

how about EnvironmentName

Reply:
It is a config option but we need information about how a given environment is configured in order to accurately promote/demote snapshots and produce correct qualified view names.

Reply:
The naming is designed to match a pattern established by `SnapshotTableInfo`. It is a subset of information contained by the primary class (`Environment` in this class and `Snapshot` in example) and the name designates it's purpose (for determining naming). `EnvironmentName` is shorter but looses the connection the established naming pattern.